### ISSUE TITLE
ALX:  a function that prints numbers, followed by a new line (variadic function)

### ISSUE THREAD

**Issue by ahmed149** [ahmed149/learning-programing-in-C#73]
#include "variadic_functions.h"
#include <stdarg.h>
#include <stdio.h>

/**
 * print_numbers - function to print numbers
 * @separator: string to separate numbers
 * @n: number of arguments
 * Return: void
 */
void print_numbers(const char *separator, const unsigned int n, ...)
{
	unsigned int a;
	va_list list;

	va_start(list, n);
	for (a = 0; a < n; a++)
	{
		printf("%d", va_arg(list, unsigned int));
		if (a != (n - 1) && separator != 0)
		{
			printf("%s", separator);
		}
	}
	printf("\n");
	va_end(list);
}

**Comment by ahmed149** [comment]
This C function, `print_numbers`, is a variadic function that prints a variable number of integers to the standard output. It also takes a separator string to insert between the printed numbers. Here's an explanation of each part of the function:

1. `#include "variadic_functions.h"` and `#include <stdarg.h>`:
   - These lines include the user-defined header file "variadic_functions.h" and the standard C library `<stdarg.h>`. The latter is essential for working with variable-length argument lists.

2. `#include <stdio.h>`:
   - This line includes the standard C library `<stdio.h>` for input and output operations, as the function uses `printf` to print the numbers.

3. `void print_numbers(const char *separator, const unsigned int n, ...)`:
   - This is the function declaration for `print_numbers`. It's defined to take a variable number of arguments represented by `...`. It also takes two additional parameters:
     - `separator`: A string used to separate the printed numbers.
     - `n`: The number of expected integer arguments.

4. `unsigned int a;`:
   - This line declares an unsigned integer variable `a`, which will be used as a loop counter.

5. `va_list list;`:
   - This line declares a `va_list` named `list`, which will be used to manage the variable-length argument list.

6. `va_start(list, n);`:
   - This line initializes the `list` variable to point to the first unnamed argument after `n`. In other words, it prepares to traverse the variable-length argument list.

7. `for (a = 0; a < n; a++)`:
   - This loop iterates `n` times, where `n` is the number of expected arguments.

8. `printf("%d", va_arg(list, unsigned int));`:
   - Inside the loop, it uses `va_arg` to retrieve the next argument from the variable-length list (`list`) as an unsigned integer and prints it using `printf`.

9. `if (a != (n - 1) && separator != 0)`:
   - This condition checks whether the current iteration is not the last one (`a != (n - 1)`) and whether a non-null separator string is provided (`separator != 0`).

10. `printf("%s", separator);`:
    - If the condition is met, it prints the separator string using `printf`.

11. `printf("\n");`:
    - After processing all the numbers and separators, it prints a newline character to end the line of output.

12. `va_end(list);`:
    - Finally, this line cleans up the `list` variable, signaling that you've finished working with the variable-length argument list.

In summary, this function allows you to pass a variable number of integers to it, along with a separator string. It then prints the integers with the specified separator between them. It uses `va_start`, `va_arg`, and `va_end` to access the variable-length argument list safely and checks for the presence of a separator string.
